requests/models.py

```diff
@@ -796,6 +797,12 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             if delimiter:
                 lines = chunk.split(delimiter)
             else:
+                # Edge case: if the last chunk ends with '\r', and the current chunk starts with \n,
+                # they should be merged and treated as only "one" new line separator '\r\n'.
+                # So the first '\n' in this chunk should be skipped since it's just the second half of 
+                # the CRLF pair ('\r\n') rather than another new line break.
```

Comment:
We should expand this comment to explain that this only affects the `splitlines` case because `splitlines` will treat any of `\r`, `\r\n`, and `\n` as newlines, and so splitting `\r\n` into two chunks will get misleading results.